coderd/httpmw/logger.go

```diff
@@ -74,3 +85,37 @@ func Logger(log slog.Logger) func(next http.Handler) http.Handler {
 		})
 	}
 }
+
+type RequestLoggerContext struct {
+	log     *slog.Logger
+	written bool
+	message string
+}
+
+func (c *RequestLoggerContext) WithFields(fields ...slog.Field) {
+	newLogger := c.log.With(fields...)
+	c.log = &newLogger
+}
+
+func (c *RequestLoggerContext) WriteLog(ctx context.Context, status int) {
+	if c.written {
+		return
+	}
+	c.written = true
+
+	if status >= http.StatusInternalServerError {
+		c.log.Error(ctx, c.message, "status_code", status)
+	} else {
+		c.log.Debug(ctx, c.message, "status_code", status)
+	}
+}
+
+type logContextKey struct{}
+
+func FromContext(ctx context.Context) *RequestLoggerContext {
```

Comment:
The function name should be more descriptive like `RequestLoggerFromContext()` perhaps, there's a lot of functions in the `httpmw` that read stuff from the context